src/core/routing.js

```diff
@@ -94,13 +94,13 @@ module.exports = class Routing {
                 // Just send it to the adapter, the talent is attached to
                 const publishOptions = ProtocolGateway.createPublishOptions(false, config.$adapterId);
 
-                // Remove these two fields from the given event
-                ev.$feature = undefined;
-                ev.$metadata = undefined;
-
                 await this.pg.publishJson(Talent.getTalentTopic(id, ev.value.$tsuffix), Object.assign(
+                    {},
                     ev,
```

Comment:
Maybe `ev` should be [frozen](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) or otherwise be made immutable in order to prevent these type of bugs.

Also, do we have a new unit test catching this bug now?

Reply:
I don't see any routing.spec file so I assume they are missing. (Is it planned for later?)

Reply:
Yes, routing spec is coming up. @argerus Good point with Object freezing to make ev being immutable in the routing component is a good idea (Where the same event is used multiple times). In other components like encoding, where an event is processed once, we are currently enriching this event by several properties. So freezing in this very case avoids this nasty error in the future in the routing component.

Reply:
I will implement that in another PR